vunit/sim_if/xcelium.py

```diff
@@ -0,0 +1,694 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+#
+# Copyright (c) 2014-2021, Lars Asplund lars.anders.asplund@gmail.com
+
+"""
+Interface for the Cadence Xcelium simulator
+"""
+
+from pathlib import Path
+from os.path import relpath
+from typing import Dict, List
+from copy import copy
+import sys
+import os
+import stat
+import subprocess
+import logging
+from ..exceptions import CompileError
+from ..ostools import write_file, file_exists, simplify_path
+from ..vhdl_standard import VHDL
+from . import SimulatorInterface, run_command, ListOfStringOption, check_output
+from .cds_file import CDSFile
+from ..color_printer import NO_COLOR_PRINTER
+from ..configuration import Configuration
+from ..project import Project
+from vunit.source_file import SourceFile
+from vunit.library import Library
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+class XceliumInterface(  # pylint: disable=too-many-instance-attributes
+    SimulatorInterface
+):
+    """
+    Interface for the Cadence Xcelium simulator
+    """
+
+    name = "xcelium"
+    supports_gui_flag = True
+    package_users_depend_on_bodies = False
+    incremental = False
+
+    compile_options = [
+        ListOfStringOption("xcelium.xrun_vhdl_flags"),
+        ListOfStringOption("xcelium.xrun_verilog_flags"),
+        ListOfStringOption("xcelium.xrun_cdslib"),
+    ]
+
+    sim_options = [ListOfStringOption("xcelium.xrun_sim_flags"),
+                   ListOfStringOption("xcelium.xrun_sim_flags.gui")]
+
+    _global_compile_options: Dict
+
+    @staticmethod
+    def add_arguments(parser):
+        """
+        Add command line arguments
+        """
+        group = parser.add_argument_group(
+            "Xcelium/Incisive", description="Xcelium/Incisive specific flags"
+        )
+        group.add_argument(
+            "--cdslib",
+            default=None,
+            help="The cds.lib file to use. If not given, VUnit maintains its own cds.lib file.",
+        )
+        group.add_argument(
+            "--hdlvar",
+            default=None,
+            help="The hdl.var file to use. If not given, VUnit does not use a hdl.var file.",
+        )
+        group.add_argument(
+            "--use_indago",
```

Comment:
```suggestion
            "--use-indago",
```
Python tools generally don't use underscores in options.

Reply:
I don't think VUnit does either.